### PR TITLE
Support MODULEDIR schema

### PR DESCRIPTION
**Pull request by adashen**
1. Upgrade the version of new create deployment template to 2.0.0
2. Parse MODULEDIR<> schema to get the external module.json
3. Clean the logic to support submodule/external module

### CODE REVIEW

File: assets/solution/deployment.template.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "$schema-template": "1.0.0",
+  "$schema-template": "2.0.0",
```
Comment (SLdragon) [inline assets/solution/deployment.template.json]: Is this version related to the schemastore json schema version? Now our latest version is 1.1.0

**Comment by formulahendry** [comment]
Shall we also add IntelliSense for MODULEDIR?
image

**Comment by adashen** [comment]
> Shall we also add IntelliSense for MODULEDIR?
> image

Since we have no control of external module path. So it is not shown. However, if the user has already added external MODULEDIR, it will be shown here.

File: assets/solution/deployment.template.json (hunk 1 above)
Comment (adashen) [reply assets/solution/deployment.template.json]: Discussed offline. Schemastore will have a new version.
